src/components/shared/Swiper.tsx

```diff
@@ -0,0 +1,329 @@
+      return;
+    }
+    const {
+        contentOffset: { x: newOffset }
+      } = event.nativeEvent,
+      { children } = this.props,
+      { index } = this.state,
+      { offset } = this.state;
+
+    // Update internal isScrolling state
+    // if swiped right on the last slide
+    // or left on the first one
+    if (
+      offset === newOffset &&
+      (index === 0 || index === children.length - 1)
+    ) {
+      this.setState({
+        isScrolling: false
+      });
+    }
+  };
+
+  /**
+   * Update index after scroll
+   * @param {object} offset content offset
+   */
+  updateIndex = (offset: number) => {
+    const state = this.state,
+      diff = offset - this.state.offset,
+      step = state.width;
+    let index = state.index;
+
+    // Do nothing if offset didn't change
+    if (!diff) {
+      return;
+    }
+
+    // Make sure index is always an integer
+    index = index + Math.round(diff / step);
+
+    // Update internal offset
+    this.setState({
+      offset: offset
+    });
+    // Update index in the state
+    this.setState({
+      index
+    });
+  };
+
+  /**
+   * Swipe one slide forward
+   */
+  swipe = () => {
+    // Ignore if already scrolling or if there is less than 2 slides
+    if (this.state.isScrolling || this.state.total < 2) {
+      return;
+    }
+
+    const state = this.state,
+      diff = this.state.index + 1,
+      x = diff * state.width,
+      y = 0;
+
+    // Call scrollTo on scrollView component to perform the swipe
+    this.scrollView && this.scrollView.scrollTo({ x, y, animated: true });
+
+    // Update internal scroll state
+    this.setState({
+      isScrolling: true
+    });
+
+    // Trigger onScrollEnd manually on android
+    // TODO: Revisit for Android https://github.com/facebook/react-native/issues/11693
+    // if (Platform.OS === "android") {
+    //   setImmediate(() => {
+    //     this.onScrollEnd({
+    //       nativeEvent: {
+    //         position: diff
+    //       }
+    //     });
+    //   });
+    // }
+  };
+
+  /**
+   * Render ScrollView component
+   * @param {array} slides to swipe through
+   */
+  renderScrollView = (pages: React.ReactNode[]) => {
+    return (
+      <ScrollView
+        ref={component => {
+          this.scrollView = component;
+        }}
+        {...this.props}
+        onScrollBeginDrag={this.onScrollBegin}
+        onMomentumScrollEnd={this.onScrollEnd}
+        onScrollEndDrag={this.onScrollEndDrag}
+      >
+        {pages.map((page, i) => (
+          // Render each slide inside a View
+          <View style={[styles.fullScreen, styles.slide]} key={i}>
+            {page}
+          </View>
+        ))}
+      </ScrollView>
+    );
+  };
+
+  /**
+   * Render pagination indicators
+   */
+  renderPagination = () => {
+    if (this.state.total <= 1) {
+      return null;
+    }
+
+    const ActiveDot = <View style={[styles.dot, styles.activeDot]} />,
```

Comment:
I think cloning these elements is kinda strange React code, I'd probably do it this way:

```
renderPagination = () => {
  if (this.state.total <= 1) {
    return null;
  }

  // this weird Array crap is just a way to generate a range from `0` to `(this.state.total - 1)` inclusive, maybe we 
  // can put that in a helper module
  const dots = [...Array(this.state.total)].keys()].map(index => {
    const styles = [styles.dot, ...(index === this.state.total ? [styles.activeDot] : [])];
    return <View style={styles} />
  });

  return <View pointerEvents="none" style={[styles.pagination, styles.fullScreen]}>{dots}</View>
}
```